tests/test_tagging.py

```diff
@@ -0,0 +1,301 @@
+import lsst.utils.tests
+from lsst.ctrl.oods.directoryScanner import DirectoryScanner
+from lsst.ctrl.oods.fileIngester import FileIngester
+from lsst.daf.butler import Butler
+from lsst.daf.butler.registry import CollectionType
+
+
+class TaggingTestCase(asynctest.TestCase):
+    """Test TAGGED deletion
+
+    This test simulates the OODS asyncio cleanup and a secondary associate
+    (tagging) and disassociate (untagging) of files, to see be sure the OODS
+    cleanup behaves properly:
+
+    1) When a dataset is TAGGED, will not be deleted, and the OODS cleaup
+       routine bypasses it.
+    2) When a dataset is not TAGGED, it can be deleted, and the OODS cleaup
+       routine removes it.
+
+    The test simulates this by gathering all ingester asyncio tasks, plus unit
+    test tasks to associate, disassociate and check for a dataset's existance
+    (or non-existance), and an interrupt task. Because the ingester cleanup
+    tasks run at predetermined intervals, the various checks in the unit tests
+    are also set up as asyncio tasks, waiting an appropriate amount of time
+    for the ingest cleanup routines to run. The code below might be a little
+    hard to follow, so here's a description of how the tasks run in this unit
+    test.
+
+    1) File ingest runs
+    2) Ingester cleanup runs, and nothing happens because the file hasn't
+       expired yet, and goes to sleep
+    3) Task to associate the dataset runs, and tags the file, and completes
+    4) Ingester cleanup task runs, finds an expired file, but doesn't deleted
+       it because it's TAGGED, and goes back to sleep
+    5) Task disassociate the dataset runs, and removes the TAGGED designation,
+        and completes
+    6) Task to check that the file runs, affirming it's still on disk, and
+       completes
+    7) Ingester cleanup task runs, finds an expired file, and deletes it,
+       since it's not TAGGED anymore and goes back to sleep
+    8) Task to check that the file runs, affirming it is not longer on disk,
+       and completes
+    7) Ingester cleanup task runs, finds nothing to do, and goes back to sleep
+    9) Task to interrupt all tasks runs, causes an exception on purpose, which
+       interrupts that gather() causing all tasks to stop.  End of unit test
+    """
+
+    async def stage(self):
+        """stage test data and set up ingester tasks
+
+        Returns
+        -------
+        staged_file: `str`
+            Path to file to after it has been staged for ingestion
+        task_list: `list`
+            A list of tasks to run for this butler ingester (ingest, cleanup)
+        """
+
+        # fits file to ingest
+        fits_name = "3019053000001-R22-S00-det000.fits.fz"
+
+        # configuration file to load
+        config_name = "ingest_tag_test.yaml"
+
+        # create a path to the configuration file
+
+        testdir = os.path.abspath(os.path.dirname(__file__))
+        configFile = os.path.join(testdir, "etc", config_name)
```

Comment:
Can we avoid mixing different naming conventions (snake_case vs camelCase) when naming variables?

Reply:
Fixed